packages/cli/src/api/workflowStats.api.ts

```diff
@@ -0,0 +1,171 @@
+import { User } from '@/databases/entities/User';
+import { whereClause } from '@/UserManagement/UserManagementHelper';
+import express from 'express';
+import { LoggerProxy } from 'n8n-workflow';
+import {
+	Db,
+	IWorkflowStatisticsCounts,
+	IWorkflowStatisticsDataLoaded,
+	IWorkflowStatisticsTimestamps,
+	ResponseHelper,
+} from '..';
+import { StatisticsNames } from '../databases/entities/WorkflowStatistics';
+import { getLogger } from '../Logger';
+import { ExecutionRequest } from '../requests';
+
+export const workflowStatsController = express.Router();
+
+// Helper function that validates the ID, throws an error if not valud
+async function checkWorkflowId(workflowId: string, user: User): Promise<void> {
+	// Check permissions
+	const shared = await Db.collections.SharedWorkflow.findOne({
+		relations: ['workflow'],
+		where: whereClause({
+			user,
+			entityType: 'workflow',
+			entityId: workflowId,
+		}),
+	});
+
+	if (!shared) {
+		LoggerProxy.info('User attempted to read a workflow without permissions', {
+			workflowId,
+			userId: user.id,
+		});
+		throw new ResponseHelper.NotFoundError(`Workflow with ID "${workflowId}" could not be found.`);
+	}
+}
+
+/**
+ * Initialise Logger if needed, and check the workflowId is acceptable
+ */
+workflowStatsController.use(async (req: ExecutionRequest.Get, res, next) => {
+	try {
+		LoggerProxy.getInstance();
+	} catch (error) {
+		LoggerProxy.init(getLogger());
+	}
+
+	// Call the checkWorkflowId function here
+	await checkWorkflowId(req.params.id, req.user);
```

Comment:
We can add this to another `workflowStatsController.use` call, so we have two separate middlewares: one for the logger and one for permissions.

One important note is that the current implementation does not prevent access, if it errors then it's simply ignored and I can access statistics about any workflow. We should instead use `res.send(...)` and prevent calling `next()` when there is any issue.